src/java.base/share/classes/java/lang/Enum.java

```diff
@@ -166,13 +168,29 @@ public final boolean equals(Object other) {
         return this==other;
     }
 
+    /**
+     * The hash code of this enumeration constant.
+     *
+     * @implNote Once initialized, the field value does not change.
+     * Hotspot's identity hash code generation also never returns zero
+     * as the identity hash code. This allows to treat zero as the marker
+     * for the un-initialized value for both {#link @Stable} and the lazy
+     * initialization code.
+     */
+    @Stable
+    private int hash;
```

Comment:
Should not we hide the field from reflection?

I understand this is an implementation detail, however, it makes a big difference that you add a private field to the *public* class that is *meant* to be inherited by user classes. Although I'm not aware of a particular example, this can be a breaking change for user code that does introspection on enums.

Reply:
How is this a big difference? It's only discovered by getDeclaredFields. This can only be modified if users break jdk internal encapsulation in command line, which means they can change everything else.